tests/geometry/test-aabb.cpp

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2019 - 2020 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/basic/logger.h>
+
+#include <geode/geometry/aabb.h>
+#include <geode/geometry/point.h>
+#include <geode/geometry/vector.h>
+
+#include <geode/tests/common.h>
+
+using namespace geode;
+
+template < index_t dimension >
+BoundingBox< dimension > create_bounding_box(
+    const Point< dimension >& center, double box_size )
+{
+    Point< dimension > temp;
+    for( index_t c : Range( dimension ) )
+    {
+        temp.set_value( c, box_size );
+    }
+    BoundingBox< dimension > box;
+    box.add_point( center - temp );
+    box.add_point( center + temp );
+
+    return box;
+}
+
+template < index_t dimension >
+std::vector< geode::BoundingBox< dimension > > create_box_vector(
+    index_t nb_box_range, double box_size )
+{
+    std::vector< geode::BoundingBox< dimension > > box_vector;
+    for( geode::index_t i : Range( nb_box_range ) )
+    {
+        for( index_t j : Range( nb_box_range ) )
+        {
+            Point< dimension > point;
+            point.set_value( 0, (double) i );
+            point.set_value( 1, (double) j );
+            box_vector.push_back( create_bounding_box( point, box_size ) );
+        }
+    }
+    return box_vector;
+}
```

Comment:
```suggestion
template < index_t dimension >
std::vector< BoundingBox< dimension > > create_box_vector(
    index_t nb_box_range, double box_size )
{
    absl::FixedArray< BoundingBox< dimension > > box_vector(nb_box_range * nb_box_range);
    for( const index_t i : Range( nb_box_range ) )
    {
        for( const index_t j : Range( nb_box_range ) )
        {
            Point< dimension > point;
            point.set_value( 0, (double) i );
            point.set_value( 1, (double) j );
            box_vector[nb_box_range * i + j] =create_bounding_box( point, box_size );
        }
    }
    return box_vector;
}
```
are (double) for casting mandatory?

Reply:
box_vector[nb_box_range * i + j] or use global_box_index just below